src/Lib/Managers/ItemsManager.vala

```diff
@@ -0,0 +1,205 @@
+
+            case InsertType.TEXT:
+                new_item = add_text (event);
+                break;
+
+            default:
+                new_item = null;
+                break;
+        }
+
+        if (new_item != null) {
+            items.append (new_item);
+        }
+
+        return new_item;
+    }
+
+    public Models.CanvasItem add_rect (Gdk.EventButton event) {
+        var rect = new Models.CanvasRect (
+            event.x,
+            event.y,
+            0.0,
+            0.0,
+            border_size,
+            border_color,
+            fill_color,
+            root
+        );
+
+        /*
+        var artboard = window.main_window.right_sidebar.layers_panel.artboard;
+        var layer = new Akira.Layouts.Partials.Layer (
+            window,
+            artboard,
+            rect,
+            "Rectangle",
+            "shape-rectangle-symbolic",
+            false
+        );
+
+        rect.set_data<Akira.Layouts.Partials.Layer?> ("layer", layer);
+
+        artboard.container.add (layer);
+        artboard.show_all ();
+        */
+
+        return rect;
+    }
+
+    public Models.CanvasEllipse add_ellipse (Gdk.EventButton event) {
+        var ellipse = new Models.CanvasEllipse (
+            event.x,
+            event.y,
+            1,
+            1,
+            border_size,
+            border_color,
+            fill_color,
+            root
+        );
+
+        /*
+        var artboard = window.main_window.right_sidebar.layers_panel.artboard;
+        var layer = new Akira.Layouts.Partials.Layer (window, artboard, ellipse,
+            "Circle", "shape-circle-symbolic", false);
+        ellipse.set_data<Akira.Layouts.Partials.Layer?> ("layer", layer);
+        artboard.container.add (layer);
+        artboard.show_all ();
+        */
+
+        return ellipse;
+    }
+
+    public Models.CanvasText add_text (Gdk.EventButton event) {
+        var text = new Models.CanvasText (
+            "Add text here",
+            event.x,
+            event.y,
+            200,
+            25f,
+            Goo.CanvasAnchorType.NW,
+            "Open Sans 18",
+            root
+        );
+
+        /*
+        var artboard = window.main_window.right_sidebar.layers_panel.artboard;
+        var layer = new Akira.Layouts.Partials.Layer (window, artboard, text, "Text", "shape-text-symbolic", false);
+        text.set_data<Akira.Layouts.Partials.Layer?> ("layer", layer);
+        artboard.container.add (layer);
+        artboard.show_all ();
+        */
+
+        return text;
+    }
+
+    private void set_item_to_insert (string type) {
+        switch (type) {
+            case "rectangle":
+                insert_type = InsertType.RECT;
+                break;
+
+            case "ellipse":
+                insert_type = InsertType.ELLIPSE;
+                break;
+
+            case "text":
+                insert_type = InsertType.TEXT;
+                break;
+        }
+    }
+
```

Comment:
These entire file looks a bit convoluted.
Why do we need the `set_insert_type_from_key`, the `insert_item`, and the `set_item_to_insert` methods repeating the switch case condition?
Can we improve this to avoid code repetition?
With this approach, whenever we have to add another shape we need to remember to update multiple methods for the same thing.

Reply:
The code repetition can be removed, the reason there are 3 different switch is because it should be taken into account both addition from keyboard AND from filemenu. 

Of course things can be improved, maybe creating a single list-model with all the shapes a user can create, with both the key associated with and the action triggered from the file menu. 

The beauty of this approach is that now `ItemsManager` is the only place in which Item addition happens, so we can refactor it without changing any other file or module (given that we respect the interface we created). 

So I think we can leave this for the future, just track it down in order not to forget it. 

What do you think?